docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,487 @@
+* When later in time a device has labels updated, the device controller will queue the device to be re-evaulated,
+  for example if a change of fleet owner happens because of the label changes.
+* An annotation is added to the device in the form of `fleet-controller/resourceVersion` to track the latest device
+  template applied to the device.
+
+> [!NOTE]
+> Devices also maintain a `metadata.resourceVersion` field, this is used to track changes related to re-applying
+> the same template to a device that has changed labels, while this is an opaque field, it could for example
+> be constructed by composing the fleet/controller resourceVersion + a hash of the device labels, or a hash
+> of the fully rendered device spec.
+
+
+Full example of fleet:
+
+```yaml
+apiVersion: v1alpha1
+kind: Fleet
+metadata:
+  creationTimestamp: "2024-04-30T14:06:17Z"
+  generation: 1
+  labels: {}
+  name: optical-inspector-production-fleet
+  owner: ResourceSync/basic-nginx-r3sourcesync
+spec:
+  selector:
+    matchLabels:
+      device: optical-inspector
+  template:
+    metadata:
+      generation: 1
+    spec:
+      config:
+      - configType: InlineConfigProviderSpec
+        inline:
+          ignition:
+            version: 3.4.0
+          storage:
+            files:
+            - contents:
+                source: data:,This%20system%20is%20managed%20by%20flightctl.%0A
+              mode: 422
+              overwrite: true
+              path: /etc/motd
+        name: motd-update
+      - configType: GitConfigProviderSpec
+        gitRef:
+          path: /basic-nginx-demo/configuration
+          repository: basic-nginx-d3mo
+          targetRevision: demo
+        name: microshift-manifests
+      - configType: GitConfigProviderSpec
+        gitRef:
+          path: /configuration
+          repository: ricky-super-secrets
+          targetRevision: main
+        name: ricky-super-secrets
+      os:
+        image: quay.io/flightctl/flightctl-agent-basic-nginx:latest
+      systemd:
+        matchPatterns:
+        - microshift.service
+        - crio.service
+        - flightctl-agent.service
+status:
+  conditions:
+  - lastTransitionTime: "2024-05-09T14:20:21Z"
+    status: "False"
+    type: OverlappingSelectors
+```
+
+#### Device management within a fleet
+
+##### Removing a device from a fleet
+
+A device could be stripped from the labels associating it to a specific fleet, in such
+case the `metadata.owner` field will be cleared by the fleet controller, and the device
+will need to be manually manage or eventually associated to a different fleet/owner.
+
+##### Changing a device to a different fleet
+
+A device could be moved to a different fleet, in such case the `metadata.owner` field will be updated
+by the fleet controller, and the device will be managed by the new fleet.
+
+This can be performed by modifying device labels to match a new fleet.
```

Comment:
"by modifying device labels to match a new fleet or modifying the fleet's device selector"?